nyx_client/nyx_client/client.py

```diff
@@ -371,26 +372,44 @@ def licenses(self) -> list[str]:
         """
         return self._nyx_get(NYX_META_LICENSES_ENDPOINT)
 
+    def _data_from_response_object(self, obj: dict[str, Any]) -> Data:
+        """Constructs a new `Data` instance for an object contained in an API response."""
+        return Data(
+            name=obj["name"],
+            title=obj["title"],
+            description=obj["description"],
+            org=self.org,
+            url=obj["accessURL"],
+            content_type=obj["contentType"],
+            creator=obj["creator"],
+            categories=obj["categories"],
+            genre=obj["genre"],
+            size=obj["size"],
+            custom_metadata=(Property.from_dict(prop) for prop in obj.get("customMetadata", ())),
+        )
+
     def search(
         self,
+        text: str,
         *,
         categories: Sequence[str] = (),
         genre: str | None = None,
         creator: str | None = None,
-        text: str | None = None,
         license: str | None = None,
         content_type: str | None = None,
         subscription_state: Literal["subscribed", "all", "not-subscribed"] = "all",
         timeout: int = 3,
         local_only: bool = False,
     ) -> list[Data]:
-        """Search for new data in the Nyx system.
+        """Find products using text in the Nyx network (or local instance).
+
+        Usage of this endpoint is discouraged. Use :func:`get_data` instead, unless you want to perform a text search.
```

Comment:
Note: Added this to match what is in the API spec. Was there a reason to not warn about it here before?